api/python/quilt3/data_transfer.py

```diff
@@ -249,6 +228,34 @@ def read_file_chunks(file, chunksize=s3_transfer_config.io_chunksize):
 UPLOAD_ETAG_OPTIMIZATION_THRESHOLD = 1024
 
 
+# 8MB - same as TransferConfig().multipart_threshold - but hard-coded to guarantee it won't change.
+CHECKSUM_MULTIPART_THRESHOLD = 8 * 1024 * 1024
+
+# Maximum number of parts supported by S3
+CHECKSUM_MAX_PARTS = 10_000
+
+
+def adjust_checksum_chunksize(file_size):
+    """
+    Calculate the chunk size to be used for the checksum. It is normally 8MB,
+    but gets doubled as long as the number of parts exceeds the maximum of 10,000.
+
+    It is the same as
+    `ChunksizeAdjuster().adjust_chunksize(s3_transfer_config.multipart_chunksize, file_size)`,
+    but hard-coded to guarantee it won't change and make the current behavior a part of the API.
+    """
+    assert file_size >= CHECKSUM_MULTIPART_THRESHOLD, "file_size too small for multi-part"
+
+    chunksize = 8 * 1024 * 1024
+    num_parts = int(math.ceil(file_size / float(chunksize)))
```

Comment:
I think it can be simplified:
```suggestion
    num_parts = math.ceil(file_size / chunksize)
```